### ISSUE TITLE
md-sidenav-collapse must have ref to the md-sidenav (and more)

### ISSUE THREAD

**Issue by djensen47** [aurelia-ui-toolkits/aurelia-materialize-bridge#484]
At first glance, this seems okay. However, I'm seeing a couple of problems:

1. There is logic required for the proper operation of `md-sidenav` contained in the logic for `md-sidenav-collapse`

Which leads me to...
2. If a toolbar (or whatever else you're using to toggle the sidenav) is not defined in the same view/component as the sidenav, you cannot use `md-sidenav-collapse`.

I have a situation where I have different toolbars for different views and some of those might need to open the sidenav while others will not. I have considered using the Event Aggregator to solve this but it appears there is some important logic contained within `md-sidenav-collapse` which might be nice to move to sidenav.

**Comment by Thanood** [comment]
You're right.
The problem is: Materialize's sidenav needs a button. It's (IMO unfortunately) all initialized using that button.

While this is not a huge problem with single page apps that actually consist of a single page where everything is globally accessible, it fails with separated views.

A workaround may indeed be the `EventAggregator` with the receiver triggering a click on an invisible button. It's really not pretty, though..

**Comment by djensen47** [comment]
Yeah, I'm using an EventAggregator now but I've switched to using the MDC temporary drawer. I think the work _you_ and this project have done is great. Unfortunately, there are some weird decisions (related and unrelated to this issue) from the Materialize team making it more and more difficult to use.

**Comment by Thanood** [comment]
Thanks for the kind words. 😃 
And I think you're right again.. Some decisions are ... well, you can see that the library is a bit old maybe.

Let's hope the best for their 1.0 refactor.
